app/logical/source/url/anifty.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+class Source::URL::Anifty < Source::URL
+  attr_reader :username, :artist_hash, :work_id, :file, :work_type
+
+  def self.match?(url)
+    url.domain == "anifty.jp" || url.host == "anifty.imgix.net" || (url.host == "storage.googleapis.com" && url.path.include?("/anifty-media/"))
+  end
+
+  def site_name
+    "Anifty"
+  end
+
+  def parse
+    case [host, *path_segments]
+
+    # https://anifty.imgix.net/creation/0x961d09077b4a9f7a27f6b7ee78cb4c26f0e72c18/20d5ce5b5163a71258e1d0ee152a0347bf40c7da.png?w=660&h=660&fit=crop&crop=focalpoint&fp-x=0.76&fp-y=0.5&fp-z=1&auto=compress
+    # https://anifty.imgix.net/creation/0x961d09077b4a9f7a27f6b7ee78cb4c26f0e72c18/48b1409838cf7271413480b8533372844b9f2437.png?w=3840&q=undefined&auto=compress
+    in "anifty.imgix.net", work_type, /0x\w+/ => artist_hash, file
```

Comment:
It would be more precise here to use `/^0x\h{40}$/` to only match the expected number of hex digits.

Reply:
I didn't want to hardcode that because we had issues with sources like weibo in the past where the length of the string changed. Not sure if the length is fixed in this kind of site however.